f3/granite.go

```diff
@@ -65,22 +67,27 @@ type GMessage struct {
 	// ID of the sender/signer of this message (a miner actor ID).
 	Sender ActorID
 
-	Current SignedMessage
-	// VRF ticket for CONVERGE messages (otherwise empty byte array).
-	Ticket Ticket
+	// Current is the payload that is signed by the signature
+	Current Payload
 	// Signature by the sender's public key over Instance || Round || Step || Value.
 	Signature []byte
+	// VRF ticket for CONVERGE messages (otherwise empty byte array).
+	Ticket Ticket
 	// Justification for this message (some messages must be justified by a strong quorum of messages from some previous step).
 	Justification Justification
 }
 
 type Justification struct {
-	Payload         SignedMessage
-	QuorumSignature QuorumSignature
+	// Payload is the payload that is signed by the signature
+	Payload Payload
```

Comment:
This is the key line that I think is the justification for a separate SignedMessage/Payload type. It's sharing the definition of the four fields that must be signed in a GMessage, and thus are the payload of an aggregation of signatures.

Is it worth it? It's pretty borderline. This change overall is an improvement, but just inlining all the fields like #58 gives up very little reuse while removing significant noise.

Let's go forward with this (we can still inline later), since serialisation stuff adds more information about how things need to work here.